scripts/build/gen_bindesc_checksums.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+import sys
+import hashlib
+from elftools.elf.elffile import ELFFile
+from elftools.elf.constants import SH_FLAGS
+
+
+BINDESC_OFFSETOF_DATA = 4
+HASH_ALGORITHMS = {
+    "md5": hashlib.md5,
+}
+
+
+def set_checksums(filename):
+    with open(filename, 'rb+') as file_stream:
+        elffile = ELFFile(file_stream)
+
+        # Variable for storing the raw bytes of the image
+        image_data = b""
+
+        # Iterate over the segments to produce the image to calculate the
+        # checksums over.
+        for segment in elffile.iter_segments():
+            if segment.header.p_type != "PT_LOAD":
+                continue
+
+            if segment.header.p_filesz == 0:
+                continue
+
+            for section in elffile.iter_sections():
+                if segment.section_in_segment(section):
+                    if not (section.name == "rom_start" and segment.section_in_segment(section)):
+                        # Don't include rom_start in the checksum calculation
+                        image_data += section.data()
```

Comment:
You'd probably want to include the load address of the chunk in the hash somehow?

As you suggest, `objcopy` could be a good approach, and hashing an ihex file would give you an equivalent - where the hash operates on a "_sparse but gaps are still encoded_" basis.

... and possibly sort the chunks by their load address? (unless they are presented in-order already / is that guaranteed)